ApprienUnitySDK/Assets/Apprien/ApprienUnitySDKExampleContent/ExampleStoreUIController.cs

```diff
@@ -8,204 +8,207 @@
+
+        /// <summary>
+        /// Called after Apprien IAP ids have replaced the base IAP ids, refreshes the prices of products
+        /// </summary>
+        /// <param name="controller"></param>
+        /// <param name="extensions"></param>
+        public void OnInitialized(IStoreController controller, IExtensionProvider extensions)
+        {
+            // Overall Purchasing system, configured with products for this application.
+            _storeController = controller;
+
+            // Store specific subsystem, for accessing device-specific store features.
+            _extensionProvider = extensions;
+
+            // Refresh prices to UI
+            RefreshUI();
+        }
+
+        /// <summary>
+        /// Called when Unity Purchasing encounters an unrecoverable initialization error.
+        ///
+        /// Note that this will not be called if Internet is unavailable; Unity IAP
+        /// will attempt initialization until it becomes available.
+        /// </summary>
+        public void OnInitializeFailed(InitializationFailureReason error)
+        {
+            Debug.Log("Initialize failed");
+        }
+
+        /// <summary>
+        /// Called when a purchase completes.
+        /// 
+        /// May be called at any time after OnInitialized().
+        /// </summary>
+        /// <param name="e"></param>
+        /// <returns></returns>
+        public PurchaseProcessingResult ProcessPurchase(PurchaseEventArgs e)
+        {
+            Debug.LogError("ProcessPurchase not implemented.");
+            return PurchaseProcessingResult.Pending;
+        }
+
+        /// <summary>
+        /// Called when a purchase fails.
+        /// </summary>
+        public void OnPurchaseFailed(UnityEngine.Purchasing.Product i, PurchaseFailureReason p)
+        {
+            Debug.Log("OnPurchaseFailed not implemented.");
+        }
+
+        /// <summary>
+        /// Get the price for the given IAP from the Store
+        /// </summary>
+        public void RefreshUI()
+        {
+            var iapProducts = _storeController.products;
+            for (var i = 0; i < _apprienProducts.Length; i++)
+            {
+                var apprienProduct = _apprienProducts[i];
+                var iapApprienProduct = iapProducts.WithID(apprienProduct.ApprienVariantIAPId);
+                var iapStandardProduct = iapProducts.WithID(apprienProduct.BaseIAPId);
+
+                var apprienPrice = iapApprienProduct.metadata.localizedPriceString;
+                var standardPrice = iapStandardProduct.metadata.localizedPriceString;
+
+                StandardPriceTexts[i].text = standardPrice;
+                ApprienPriceTexts[i].text = apprienPrice;
+
+                // Update the Apprien IAP ids to text
+
+                ApprienPriceSKUTexts[i].text = apprienProduct.ApprienVariantIAPId;
+            }
+
+            // Tell Apprien that the products were shown
+            _apprienManager.ProductsShown(_apprienProducts);
```

Comment:
StartCoroutine(_apprienManager.ProductsShown(_apprienProducts));

Otherwise the request is sent but network or HTTP errors cannot be detected and further reported.